src/module_3/module_3.py

```diff
@@ -0,0 +1,129 @@
+# %% Import packages
+import pandas as pd
+
+from sklearn.model_selection import train_test_split
+from sklearn import linear_model
+from sklearn.linear_model import Ridge
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.ensemble import RandomForestClassifier
+from sklearn import metrics
+from sklearn.preprocessing import normalize
+
+import matplotlib.pyplot as plt
+
+# %% Load data
+address = "D:\\Yo\\"
+filename = "feature_frame.csv"
+df = pd.read_csv(address + filename)
+
+# %% Data preprocessing
+
+# Obtain the id's for the orders where 5 or more products where bought
+filter_df = df[df["outcome"]==1]
+filter_df = filter_df.groupby("order_id").filter(lambda x: len(x) > 4)
+
+# Filter the data so we retain only those ids
+dataset = df[df["order_id"].isin(filter_df["order_id"])]
+
+# Change date format to retain only year, month and day
+dataset["created_at"] = dataset["created_at"].str[:-9]
+dataset["created_at"] = dataset["created_at"].str.replace("-","")
+pd.to_numeric(dataset["created_at"])
+dataset["order_date"] = dataset["order_date"].str[:-9]
+dataset["order_date"] = dataset["order_date"].str.replace("-","")
+pd.to_numeric(dataset["order_date"])
+
+# Order by date
+dataset = dataset.sort_values(by=["order_date"])
+
+# %% Drop qualitative features
+print(dataset.columns)
+print(dataset.dtypes)
+dataset.drop(["product_type" , "vendor"], axis=1, inplace=True)
+
+# %% Divide the data in training, validation and test set
+def timesplit_by_order_id(data: pd.DataFrame, train_size: float, validation_test_size: float):
+    total_orders = data.order_id.unique()
+    train_num = round(total_orders.size*train_size)
+    val_num = round(total_orders.size*validation_test_size)
+    test_num = val_num
+    gap = round(total_orders.size*(1-validation_test_size-train_size)/3)
+
+    train_orders=total_orders[0:train_num]
+    val_orders=total_orders[train_num+gap:train_num+gap+val_num]
+    test_orders=total_orders[train_num+2*gap+val_num:train_num+2*gap+val_num+test_num]
+
+    train = dataset[dataset["order_id"].isin(train_orders)]
+    val = dataset[dataset["order_id"].isin(val_orders)]
+    test = dataset[dataset["order_id"].isin(test_orders)]
+
+    return train, val, test
+
+train_set, val_set, test_set = timesplit_by_order_id(dataset, 0.74, 0.10)
```

Comment:
Okey reading the name of the function is not clear to me if you are trying to do a random split or a split by time (I added [this](https://github.com/geeeb95/zrive-ds/pull/2/files#r1393410254) comment thinking that you wanted to do a random split) . If you do want to do a split by time I don't think is as easy as sorting the orders in chronological order and then taking the first chunk as train, the second as eval and the last as test. The reason is that you don't know how these orders are distributed over time: imagine that the first orders are very spread apart while the last ones (the most recent ones) are happening in a very small period of time. This would mean that, in a very extreme case, the orders of eval and test may happen even in the same day, while train ones may happen over several months, which doesn't seem like a great split. Instead, what one normally does is to split the time itself: for example, train would be all the orders happening from Jan to August (independently on how many orders are them), eval would be the orders from Sept to Oct and test the ones from November to December. One also normally does some EDA to understand how the orders are spread over time